sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/batch/BulkExecutor.java

```diff
@@ -887,7 +890,20 @@ private Mono<CosmosBatchResponse> executeBatchRequest(PartitionKeyRangeServerBat
 
         return withContext(context -> {
             final Mono<CosmosBatchResponse> responseMono = this.docClientWrapper.executeBatchRequest(
-                BridgeInternal.getLink(this.container), serverRequest, options, false);
+                BridgeInternal.getLink(this.container), serverRequest, options, false)
+                .flatMap(cosmosBatchResponse -> {
```

Comment:
IMO this flatMap should not allow concurrency >1?

Reply:
The publisher is of type `Mono<T>` and not `Flux<T>` so concurrency wouldn't apply here.